### PR TITLE
Add sitemap support

### PR DESCRIPTION
**Pull request by xcffl**
Also adds vscode configs into .gitignore.

### CODE REVIEW

File: cps/admin.py [hunk 1]
```diff
@@ -569,7 +569,12 @@ def _configuration_update_helper():
     # Remote login configuration
     _config_checkbox("config_remote_login")
     if not config.config_remote_login:
-        ub.session.query(ub.RemoteAuthToken).filter(ub.RemoteAuthToken.token_type==0).delete()
+        ub.session.query(ub.RemoteAuthToken).filter(ub.RemoteAuthToken.token_type == 0).delete()
+
```
Comment (OzzieIsaacs) [inline cps/admin.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/Ozzie-IsaacsOrganization/calibre-web/pullRequest?prid=4985812)

**Comment by OzzieIsaacs** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- cps/admin.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ozzie-IsaacsOrganization/calibre-web/pullRequest?prid=4985812&bid=16462786)

**Comment by OzzieIsaacs** [comment]
Thank you for your contribution. Meanwhile I had time to take a look at your code. 

For me it isn't working, I got an empty xml feed(? what did I do wrong, what should I do?) 
I currently don't see the benefit of having a sitemap for calibre-web (it adds another dependency and complexity) 
So my question, what is the purpose/benefit of the sitemap?

**Comment by xcffl** [comment]
It's for search engines to do indexing if you want your website to be exposed to the public.

I'll check the code later. I guess there is a small bug.

**Comment by vagra** [comment]
I added my website to the search engine. It has a lot of search count and 500+ register users, So there is indeed a need to generate a sitemap. Currently, I use a script to automatically generate a sitemap

**Pull request closed without merging**